### DISCUSSION TITLE
Add event callback prop to Auth UI component

### DISCUSSION THREAD

**Discussion by 0xturner** [supabase/supabase#13219] [Feature Requests]
It would be handy if the `Auth` component received an event callback prop so that developers could trigger actions when the internal state changes.

The usage in React might look something like this:
```
<Auth
    ...
    onAuthEvent={(event) => {
        if (event === "sign_in") {
            foo();
        } else if (event === "sign_up") {
            bar();
        }
    }}
/>
```

My specific use case: When a user signs up I would like to hide the `Auth` component and display a custom component view in its place.

I would be happy to submit a PR for this feature but I would like to receive feedback on the feature idea before starting on the implementation.

**Comment by justin-gurtz** [comment]
I was about to create a similar discussion, I have a use case where I need to know the Auth UI `view` state in order to update surrounding components.

(By the way this seems similar to https://github.com/orgs/supabase/discussions/14646)
